source/citnames/source/Output.cc

```diff
@@ -92,37 +92,40 @@ namespace {
     // maintains the hash sets). If an entry has no output attribute, then it switch to use
     // all attributes hashes.
     struct DuplicateFilter : public Filter {
-        DuplicateFilter()
-                : hashes()
+        DuplicateFilter(bool strict_duplicate)
```

Comment:
Instead of a `bool` parameter, I would suggest to create another class which implements the `Filter`. Even further, maybe create a `DuplicateFilter` class, which expects a `virtual std::string hash(const cs::Entry &entry) const = 0;` method? And that would have two implementations? And create a factory method on the `DuplicateFilter` class (as static method), so the whole duplicate filter business is encapsulated there.